### DISCUSSION TITLE
Hot Reload support

### DISCUSSION THREAD

**Discussion by civet** [airsdk/Adobe-Runtime-Support#2782] [Ideas]
As we all know, many web development environments today support hot reloading, thanks to the browser's mechanism for executing JS without precompilation.

For a language like AS3 that needs to be pre-compiled before it can be executed at runtime, it seems that there is no consideration for implementing a workflow that can quickly preview the results as long as the code is modified.

I don’t need to talk about the [benefits](https://www.youtube.com/watch?v=EGqwXt90ZqA) of supporting hot reloading. I just want to know if AS3 can also support it. After all, it’s 2023 now 😂

Back in 2013, there was a tool called [COLT](https://www.youtube.com/watch?v=LdE5QWHvB-Q) that did it, but it's been **abandoned** for years. Just leave the code repository [here](https://github.com/code-orchestra)

**Comment by ajwfrost** [comment]
Hi

We'd put in place a change a while ago that laid some of the groundwork for this, and had done a bit of a proof of concept of it.. but ultimately it will also require an update to FDB and to the IDEs in order to support this...

In terms of configuring this perhaps, we would need to have a flag somewhere to set it up (or, it could just become the default behaviour when you use a particular version of the AIR SDK). Then it relies on there being an FDB connection to the AIR runtime; FDB would have to be monitoring the original SWF file, and when this changes, the new SWF would be pushed up to the runtime and then the behaviour would need to change (we throw away any JITted functions and they'd just get re-generated on next call).

At a simple level, the whole SWF could be uploaded and the app restarted, but it may also be possible to detect changes on a function-by-function level and just update those.. although there may be challenges then with some of the references within the byte code, and if any items in the constant pool had changed then the whole SWF would need to be re-started.. so it's not just the fact that we switch into interpreter mode, there are a lot of complexities due to the fact that SWF files contain ActionScript Byte Code rather than ActionScript itself...

An interesting challenge though..! Definitely one that we want to progress..

**Reply by civet** [reply]
Thank you for the detailed explanation. I don't quite understand the deeper mechanism. It is good to know that there has been progress in this feature. it doesn't need to be perfect, maybe it will happen someday. We look forward to it!

 [Makc](https://github.com/makc) is a genius in the COLT development team. He blogged about [it](https://makc3d.wordpress.com/tag/colt/).
I hope this information helps.
